File: 3 semestre/Aplicações para desktop/LivroIHM/LivroIHM/LivroBLL.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LivroIHM
{
    internal class LivroBLL
    {
        public void validaDados(Livro livro)
        {
            if (livro.getTitulo.Length == 0)
            {
                Erro.setErro("O campo de título deve ser preenchido");
                return;
            }

            if (livro.getAutor.Length == 0)
            {
                Erro.setErro("O campo de título deve ser preenchido");
                return;
            }

            if (livro.getEditora.Length == 0)
            {
                Erro.setErro("O campo de título deve ser preenchido");
                return;
            }

            if (livro.getAnoedicao.Length == 0)
            {
                Erro.setErro("O campo de título deve ser preenchido");
                return;
            }
            else
            {
                try
                {
                    float.Parse(livro.getAnoedicao);
                }
                catch
                {
                    Erro.setErro("O campo de ano deve ser numérico");
                    return;
                }
                if (float.Parse(livro.getAnoedicao) < 0)
                {
                    Erro.setErro("o campo de ano deve não pode ser preenchido com números negativos");
                }
            }

            if (livro.getLocal.Length == 0)
            {
                Erro.setErro("O campo de título deve ser preenchido");
                return;
            }
        }
    }
}

File: 5 semestre/Técnicas de programação/firebase/src/main/java/desafio/asenjo/firebase/LivroBLL.java
package desafio.asenjo.firebase;

public class LivroBLL {
  public static void validaDados(Livro umlivro) {
    Erro.setErro(false);
    if (umlivro.getTitulo().equals("")) {
      Erro.setErro("O campo TITULO é de preenchimento obrigatório...");
      return;
    }
    if (umlivro.getAutor().equals("")) {
      Erro.setErro("O campo AUTOR é de preenchimento obrigatório...");
      return;
    }
    if (umlivro.getEditora().equals("")) {
      Erro.setErro("O campo EDITORA é de preenchimento obrigatório...");
      return;
    }
    if (umlivro.getAnoEdicao().equals("")) {
      Erro.setErro("O campo ANO EDICAO é de preenchimento obrigatório...");
      return;
    } else
      try {
        Integer.parseInt(umlivro.getAnoEdicao());
      } catch (Exception e) {
        Erro.setErro("O campo ANO EDICAO deve ser numerico!");
        return;
      }
    if (umlivro.getLocalizacao().equals("")) {
      Erro.setErro("O campo LOCALIZACAO é de preenchimento obrigatório...");
      return;
    }
    LivroDAL.conecta();
    if (Erro.getErro())
      return;
    LivroDAL.inseriLivro(umlivro);
  }

  public static void validaTitulo(Livro umlivro) {
    Erro.setErro(false);
    if (umlivro.getTitulo().equals("")) {
      Erro.setErro("O campo TITULO é de preenchimento obrigatório...");
      return;
    }
    LivroDAL.conecta();
    if (Erro.getErro())
      return;
    LivroDAL.consultaLivro(umlivro);
    if (Erro.getErro())
      return;
  }
}
